hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecord.java

```diff
@@ -18,21 +18,21 @@
 
 package org.apache.hudi.common.model;
 
-import java.util.Map;
-import java.util.stream.Collectors;
-import java.util.stream.IntStream;
 import org.apache.hudi.common.util.CollectionUtils;
 import org.apache.hudi.common.util.Option;
+import org.apache.hudi.common.util.collection.Pair;
 
 import java.io.Serializable;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
-import org.apache.hudi.common.util.collection.Pair;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 /**
  * A Single Record managed by Hoodie.
  */
-public class HoodieRecord<T extends HoodieRecordPayload> implements Serializable {
+public abstract class HoodieRecord<T> implements Serializable {
```

Comment:
@yihua @xushiyan let's chat more on this to make sure we're aligned on the approach going f/w:

I was thinking of keeping this component file-format agnostic and instead make it engine-specific, while refactoring MOR table read-path for efficient querying. 

Can you elaborate what's the goal you're striving for w/ `HoodieAvroRecord`?

P.S. Putting this context in here for somebody who might not be aware of previous conversations

Reply:
@alexeykudinkin so the main goal here is to make `HoodieRecord` independent of `HoodieRecordPayload`, this will allow `HoodieRowRecord` to be used by row-writer code path. `HoodieAvroRecord` is also meant for compatibility with existing codebase and ease the decoupling. Yea we should discuss more about HoodieAvroRecord vs SparkHoodieRecord.

Reply:
Discussed offline: this will be an intermediate state until RFC-46 is fully implemented